database/sql/src/main/scala/cromwell/database/slick/tables/MetadataEntryComponent.scala

```diff
@@ -323,14 +323,18 @@ trait MetadataEntryComponent {
       if(isPostgres) "obj.data" else "METADATA_VALUE"
     }
 
-    def targetCallsSelectStatement(callFqn: String, scatterIndex: String, retryAttempt: String): String = {
-      s"SELECT ${callFqn}, MAX(COALESCE(${scatterIndex}, 0)) as maxScatter, MAX(COALESCE(${retryAttempt}, 0)) AS maxRetry"
+    def attemptAndIndexSelectStatement(callFqn: String, scatterIndex: String, retryAttempt: String, variablePrefix: String): String = {
+      s"SELECT ${callFqn}, MAX(COALESCE(${scatterIndex}, 0)) as ${variablePrefix}Scatter, MAX(COALESCE(${retryAttempt}, 0)) AS ${variablePrefix}Retry"
```

Comment:
Should this be `sql""` instead of `s""`? The latter does plain string interpolation and would seem to create an injection vector.

[This page](https://scala-slick.org/doc/3.4.1/sql.html#string-interpolation) says "no danger of SQL injection attacks" in reference to `sql""`/`sqlu""` specifically.

Reply:
I'm not binding values to an SQL statement but rather dynamically determining table and/or column names (depending on the situation). In these circumstances you'll have to splice the value directly, as outlined here

https://scala-slick.org/doc/3.4.1/sql.html#splicing-literal-values

As far as I know sanitization is only applied to bound variables. In the scenario where you need to splice literal values you need to ensure that it isn't unchecked user input. In this case I'm splicing variables of my own creation into the statement.

Reply:
Oh and in addition `s"` returns a type `String` whereas `sql"` returns a type `SQLActionBuilder`. Since the partial statement is using dynamic column/table names anyways the `String` is easier to work with.

Reply:
Sounds good, thanks for the explanation.